authorize/grpc.go

```diff
@@ -192,6 +192,14 @@ func (a *Authorize) handleForwardAuth(req *envoy_service_auth_v2.CheckRequest) b
 	}
 
 	uriQuery := checkURL.Query().Get("uri")
+	if headers := req.GetAttributes().GetRequest().GetHttp().GetHeaders(); uriQuery == "" && headers != nil {
```

Comment:
nit: this is a little hard to read and mentally parse.

Reply:
The `uriQuery` is used in many places, while the `headers` only used if the condition is true. I often keep the scope of variable as small as possible.